src/clients/graphql/graphql_client.ts

```diff
@@ -2,32 +2,57 @@ import {MissingRequiredArgument} from '../../error';
 import {Context} from '../../context';
 import {ShopifyHeader} from '../../base_types';
 import {HttpClient} from '../http_client/http_client';
-import {DataType, RequestReturn} from '../http_client/types';
+import {ApiClientParams, DataType, RequestReturn} from '../http_client/types';
 import * as ShopifyErrors from '../../error';
 
 import {GraphqlParams} from './types';
 
+export interface AccessTokenHeader {
+  header: string;
+  value: string;
+}
+
 export class GraphqlClient {
+  readonly domain: string;
+  readonly accessToken?: string;
+
+  protected baseApiPath = '/admin/api';
+
   private readonly client: HttpClient;
 
-  constructor(readonly domain: string, readonly token?: string) {
-    if (!Context.IS_PRIVATE_APP && !token) {
-      throw new ShopifyErrors.MissingRequiredArgument('Missing access token when creating GraphQL client');
+  // When we next release a major version for this library, we should remove string as a valid type for params and
+  // remove the accessToken param. Also remove the first block.
+  public constructor(params: ApiClientParams | string, accessToken?: string) {
```

Comment:
so the actual proposed constructor will be?
```
public constructor(params: ApiClientParams) { ... }
```
Since this is a proposal, why not just change it across the library?

Reply:
Actually, why change it to a type?
From, essentially ...
```
constructor(domain: string, token?: string)
```
...to
```
constructor({domain: string, token?: string})
```

Reply:
Yes, this isn't strictly necessary, but in most other places in the library we're using types rather than separate parameters (which makes functions easier to evolve), so I was going for consistency.

The reason I didn't make the full change is to keep this backward compatible until we need to do a major release. Current apps would be relying on the old format and I was trying to avoid breaking them if we could.

All of that said, I have no strong feelings about doing this. This is a fairly simple refactoring that we can do at any time.

Reply:
I'm rolling back this change, it should be done in a separate PR if we're not making any actual changes to constructors here any more.